### ISSUE TITLE
Consider moving accuracy to properties in the route.

### ISSUE THREAD

**Issue by aickin** [openmobilityfoundation/mobility-data-specification#51]
When a GPS device moves through an environment, especially an urban environment, the accuracy of the signal can vary tremendously. This means it's possible that some points on a GPS route may have an accuracy of a meter while others points in the route may have accuracy of 10 or 30 meters or whatever. In this kind of case, it's not entirely clear what to send in the `accuracy` field for a trip, since it varies so much.

I'd suggest that maybe it would be better to move the `accuracy` field into the `route` GeoJSON, like this:

```
"route": {
    "type": "FeatureCollection",
    "features": [{
        "type": "Feature",
        "properties": {
            "timestamp": 1529968782.421409,
            "accuracy": 1,
        },
        "geometry": {
            "type": "Point",
            "coordinates": [
                -118.46710503101347,
                33.9909333514159
            ]
        }
    },
    {
        "type": "Feature",
        "properties": {
            "timestamp": 1531007628.3774529
            "accuracy": 10,
        },
        "geometry": {
            "type": "Point",
            "coordinates": [
                -118.464851975441,
                33.990366257735
            ]
        }
    }]
}
```

**Comment by ezheidtmann** [comment]
Strongly agree; accuracy information should always be associated with a point location.

In our experience with smartphone chips, GPS location accuracy is communicated in two ways.

 * [Android](https://developer.android.com/reference/android/location/Location#getAccuracy()) - `accuracy` - "radius of 68% confidence" (although our experience shows this definition is not rigorous)
 * [iOS](https://developer.apple.com/documentation/corelocation/cllocation/1423599-horizontalaccuracy) - `horizontalAccuracy` - "radius of uncertainty for the location"

In the general case, [it doesn't appear that accuracy can be reliably computed from standard NMEA sentences](https://www.gps-forums.com/threads/estimating-accuracy-from-raw-nmea-data.46273/) but I bet most stacks can produce a radius of uncertainty.

Of course, the real location isn't always within the radius, even when the radius is very small. And both smartphone platforms have additional layers of location processing that can add errors. So we always track which platform produced a location so we can process differently if needed.

---

So I'd suggest using the field `horizontalAccuracy` for clarity, and perhaps including a GPS platform enum in the route or trip description.

```
"route": {
    "type": "FeatureCollection",
    "features": [{
        "type": "Feature",
        "properties": {
            "timestamp": 1529968782.421409,
            "horizontalAccuracy": 1,
        },
        "geometry": {
            "type": "Point",
            "coordinates": [
                -118.46710503101347,
                33.9909333514159
            ]
        }
    },
    {
        "type": "Feature",
        "properties": {
            "timestamp": 1531007628.3774529
            "horizontalAccuracy": 10,
        },
        "geometry": {
            "type": "Point",
            "coordinates": [
                -118.464851975441,
                33.990366257735
            ]
        }
    }],
    "properties": {
        "gpsProvider": "ios"
    }
}
```

Happy to prepare a PR if either of these changes sound good.

**Comment by jfh01** [comment]
Flagging #61 as a separate, but related issue.

**Comment by ezheidtmann** [comment]
@thekaveman I present the accuracy information as context from our area of expertise, and as a possible basis for a definition of accuracy to be included in the spec. (These definitions are common & I believe all providers should be able to provide an accuracy figure that is useful for analysis)

We've seen that some providers don't have GPS chips on all their devices (they rely on customer devices for location information); others run Android on the scooters. I don't know how many providers are currently uploading locations from their devices rather than relying on customer devices.

**Comment by schnuerle** [comment]
Should this be an optional field across MDS whenever location/telemetry data is used? I haven't heard much talk of this in practice with cities or providers, but if it's still a need 2 years later we can look to add it.

**Comment by thekaveman** [comment]
@schnuerle it is an optional field now for Provider `trips`. I believe this issue started as a simple suggestion to move the `accuracy` field from the top-level into the `route` object (similar to how we added `stop_id` directly to geometric data vs. at the top-level). Since the accuracy for any given point measurement may be different from a subsequent measurement (i.e. the idea that the trip itself has an accuracy measurement is somewhat... inaccurate).

**Comment by quicklywilliam** [comment]
Plus one on this. There are several other GPS fields we'd love to see besides accuracy that are available on Agency but not Provider. This includes speed, heading, the number of satellites fixed and the altitude. These data can be tremendously helpful to filtering and cleaning up GPS data.

Also, this effort to bring /trips to Agency is related (in so far as we are slowly but surely unifying all the things):
https://github.com/openmobilityfoundation/mobility-data-specification/issues/550

**Comment by quicklywilliam** [comment]
@schnuerle per our conversation today, should we tag this as a 2.0 issue? Or I can file a new one?

**Comment by schnuerle** [comment]
Thanks for reviving this issue you all. I think it can be closed in favor of a more up to date issue with a more accurate title and description that summarizes the current issue and how it could be solved.

**Comment by schnuerle** [comment]
Being addressed by #653. Please review that PR and leave comments there if interested.
